exercises/allergies/.gitignore

```diff
@@ -0,0 +1,7 @@
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
```

Comment:
i actually missed the 3rd learning moment which is the student asking "what the heck is a Cargo.lock file" which is a good thing to learn sooner than later as well!